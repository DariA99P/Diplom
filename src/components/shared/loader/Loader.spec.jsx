import React from 'react';
import { shallow } from 'enzyme';

import Loader from '.';

describe('<Loader />', () => {
  describe('On rendering', () => {
    let tree;

    beforeEach(() => {
      tree = shallow(<Loader />);
    });

    describe('When some error during loading process occurs', () => {
      beforeEach(() => {
        tree.setProps({ error: true });
      });

      it('should render something idicating it', () => {
        expect(tree).toMatchSnapshot();
      });
    });

    describe('When loading process take considerable ammount time to finish', () => {
      beforeEach(() => {
        tree.setProps({ pastDelay: true });
      });

      it('should render loading animation', () => {
        expect(tree).toMatchSnapshot();
      });
    });

    describe('When no error occurs and loading process quickly finish ', () => {
      beforeEach(() => {
        tree.setProps({ error: false, pastDelay: false });
      });

      it('should render nothing', () => {
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
