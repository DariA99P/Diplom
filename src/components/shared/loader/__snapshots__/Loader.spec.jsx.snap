// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> On rendering When loading process take considerable ammount time to finish should render loading animation 1`] = `
<div
  className="loaderWrapper"
>
  <div
    className="loader"
  />
</div>
`;

exports[`<Loader /> On rendering When no error occurs and loading process quickly finish  should render nothing 1`] = `""`;

exports[`<Loader /> On rendering When some error during loading process occurs should render something idicating it 1`] = `
<h1>
  Something Went Wrong!
</h1>
`;
